cordformation/src/main/kotlin/net/corda/plugins/Node.kt

```diff
@@ -323,16 +322,73 @@ open class Node @Inject constructor(private val project: Project) : CordformNode
                 .withValue("rpcSettings.address", ConfigValueFactory.fromAnyRef("$containerName:${rpcSettings.port}"))
                 .withValue("rpcSettings.adminAddress", ConfigValueFactory.fromAnyRef("$containerName:${rpcSettings.adminPort}"))
                 .withValue("detectPublicIp", ConfigValueFactory.fromAnyRef(false))
-        val tmpConfFile = createTempConfigFile(dockerConf.root())
+
+        createNodeAndWebServerConfigFiles(dockerConf)
+    }
+
+    private fun createNodeAndWebServerConfigFiles(config: Config) {
+
+        val tmpConfFile = createTempConfigFile(config.toNodeOnly().root(), "node.conf")
         appendOptionalConfig(tmpConfFile)
         project.copy {
             it.apply {
                 from(tmpConfFile)
                 into(rootDir)
             }
         }
+        if (config.hasPath("webAddress")) {
+            val webServerConfigFile = createTempConfigFile(config.toWebServerOnly().root(), "web-server.conf")
+            project.copy {
+                it.apply {
+                    from(webServerConfigFile)
+                    into(rootDir)
+                }
+            }
+        }
+    }
+
+    private fun Config.toNodeOnly(): Config {
+
+        return if (hasPath("webAddress")) {
+            withoutPath("webAddress").withoutPath("useHTTPS")
+        } else {
+            this
+        }
     }
 
+    private fun Config.toWebServerOnly(): Config {
+
+        var config = ConfigFactory.empty()
+        config = copyTo("webAddress", config)
+        config = copyTo("myLegalName", config)
+        if (hasPath("rpcOptions.address") || hasPath("rpcAddress")) {
+            config += "rpcAddress" to if (hasPath("rpcOptions.address")) {
+                getValue("rpcOptions.address")
+            } else {
+                getValue("rpcAddress")
+            }
+        }
+        config = copyTo("rpcUsers", config)
+        config = copyTo("useHTTPS", config)
+        config = copyTo("baseDirectory", config)
+        config = copyTo("keyStorePassword", config)
+        config = copyTo("trustStorePassword", config)
+        config = copyTo("exportJMXto", config)
+        config = copyTo("custom", config)
+        return config
+    }
+
+    private fun Config.copyTo(key: String, target: Config, targetKey: String = key): Config {
+
+        return if (hasPath(key)) {
+            target + (targetKey to getValue(key))
+        } else {
+            target
+        }
+    }
+
+    private operator fun Config.plus(property: Pair<String, Any>) = withValue(property.first, ConfigValueFactory.fromAnyRef(property.second))
```

Comment:
Can this and copyTo be moved into a Utils.kt file. These are generic enough that they don't belong in the node.